Task 2

RamanReut opened the pull request.

RamanReut committed:
Add redux store

RamanReut committed:
Complete basic layout

RamanReut committed:
Add Navbar expand button; Add breadcrumbs

RamanReut committed:
Add search field and avatar

RamanReut committed:
Add navbar; Some changes on header for better
responsive

RamanReut committed:
Add unit tests

RamanReut committed:
Rewrite some layote. Add data table

RamanReut committed:
add unit tests

RamanReut committed:
Rework styles

RamanReut committed:
Rework unit tests; Add eslinet eol-last rule

RamanReut committed:
Merge changes in task_1 to task_2

RamanReut committed:
rework unit tests

RamanReut committed:
delete unuse dependences

dHihich reviewed (changes requested):
I see all the changes from previous PR here - not good. I don't have time to review all files twice. Please create PR so that I could see changes for task2 only (task_2 -> task_1 or resolve comments for task_1 and merge it to master and recreate task_2 -> master)

dHihich commented:
please change view for column selector. It takes too much place and moreover it shouldn't be visible always Add an icon and show a popup with checkboxes on hover.

dHihich commented:
add loading indicator while loading data (now it's showing an empty grid while loading)

dHihich commented:
now all routes render Data grid. Please show a blank page for all routes except of `/data`
add 404 page for unknown routes

dHihich commented:
Is it possible to customize filter for datetime columns? I think it would be good to have a datepicker when you click on filter. Also would be good to filter by date range. Is it achievable?

![image](https://user-images.githubusercontent.com/30228677/91712016-9b811480-eb8f-11ea-9a86-99a9ba816085.png)

NOTE: no need to do this right now, you can do it later on in separate PR

RamanReut committed:
Add expand animation to search input on hover;
Extract search input into separate component;
Hide menu button for large and middle monitor

RamanReut committed:
Merge branch 'task_1' into task_2

RamanReut committed:
Rework column visibility control

RamanReut committed:
add data load animation

RamanReut committed:
Add not found page

RamanReut committed:
Add date picker for filtering date

RamanReut committed:
Update dependencies

RamanReut committed:
Minor types name fixes

RamanReut committed:
Update dependencies

RamanReut committed:
Add unit tests
